cmd/kubeadm/app/node/bootstrap.go

```diff
@@ -63,10 +59,10 @@ func EstablishMasterConnection(c *kubeadmapi.TokenDiscovery, clusterInfo *kubead
 	caCert := []byte(clusterInfo.CertificateAuthorities[0])
 
 	stopChan := make(chan struct{})
-	result := make(chan *ConnectionDetails)
+	result := make(chan *clientcmdapi.Config)
```

Comment:
It would be a bit cleaner to have this be a `var result *clientcmdapi.Config` and set it in a sync.Once